### DISCUSSION TITLE
Generation of code from SDL (without connection to the GraphQL server)

### DISCUSSION THREAD

**Discussion by obabichev** [ExpediaGroup/graphql-kotlin#911] [Q&A] [answered]
First of all, thank you for the library, I was using it during the last weeks for creating E2E API tests.

But during the usage and reading the documentation I found the only way to generate the code using the existing GraphQL server. Actually, one of gradle tasks `graphqlIntrospectSchema`, `graphqlDownloadSDL`(depending on configuration) runs always during the build, and I have not found a way to avoid it.

So the question is it possible to generate the whole client code without exiting the running GraphQL server?

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
If you are using `graphql` Gradle plugin extension then yes it is equivalent of running introspect/download SDL followed by client generation (see https://expediagroup.github.io/graphql-kotlin/docs/plugins/gradle-plugin#complete-configuration-example). There is no way to configure client generation to use local file when using the extension. Instead you will need to configure the individual task.

See https://github.com/ExpediaGroup/graphql-kotlin/discussions/905 for details.

**Reply by obabichev** [reply]
Thank you a lot, I still have to specify `endpoint` in build.gradle.kts, but it's small problem, I can put any magic value, and snipped like 
```
val graphqlGenerateClient by tasks.getting(com.expediagroup.graphql.plugin.gradle.tasks.GraphQLGenerateClientTask::class) {
    packageName.set("com.smth.generated")
    schemaFile.set(file("schema.graphql"))
}

val graphqlIntrospectSchema by tasks.getting(com.expediagroup.graphql.plugin.gradle.tasks.GraphQLIntrospectSchemaTask::class) {
    enabled = false
}
```

helped me.

**Reply by dariuszkuc** [reply]
If you configure the tasks as above then there is no need to use the extension (also if you don't use the extension you won't have to explicitly disable the introspection task either as it won't be configured).
